3.25 release blog post (#2520)
sourcegraph/about: blogposts/2021/release-3.25.md

---
title: "Sourcegraph 3.25 release"
publishDate: 2021-02-20T10:00-07:00
description: "Sourcegraph 3.25 is released."
tags: [blog, release]
slug: "release/3.25"
published: true
changelogItems:

> - description: "Go `1.15` introduced changes to SSL/TLS connection validation which requires certificates to include a `SAN`. This field was not included in older certificates and clients relied on the `CN` field. You might see an error like `x509: certificate relies on legacy Common Name field`. We recommend that customers using Sourcegraph with an external database that is connected using SSL/TLS check whether the certificate is up to date. AWS RDS customers please reference [AWS' documentation on updating the SSL/TLS certificate](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL-certificate-rotation.html) for steps to rotate your certificate."

Reviewer: @content-hacker - this is an important updated to include, but it doesn't align well with the short descriptions we normally use. Thoughts on having this be the last update, but keeping the length?

Reviewer 2: @abschu looking at the preview, the longer length isn't jarring. If it's an important update, I would keep it at the top to make sure people see it.

I think last release @Joelkw figured out that we can't externally link from this post, it breaks it. When I look at the preview, the link doesn't work and categories are missing, likely because of the link to aws.

Reviewer 3: Yep, one "problem" of this format is that you can't link in the description, but we do let people link in the changelog (which feeds this description).

> - description: "Go `1.15` introduced changes to SSL/TLS connection validation which requires certificates to include a `SAN`. This field was not included in older certificates and clients relied on the `CN` field. You might see an error like `x509: certificate relies on legacy Common Name field`. We recommend that customers using Sourcegraph with an external database and and connecting to it using SSL/TLS check whether the certificate is up to date. AWS RDS customers please reference [AWS' documentation on updating the SSL/TLS certificate](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL-certificate-rotation.html) for steps to rotate your certificate."

Reviewer: ```suggestion
- description: "Go `1.15` introduced changes to SSL/TLS connection validation which requires certificates to include a `SAN`. This field was not included in older certificates and clients relied on the `CN` field. You might see an error like `x509: certificate relies on legacy Common Name field`. We recommend that customers using Sourcegraph with an external database that is connected using SSL/TLS check whether the certificate is up to date. AWS RDS customers please reference [AWS' documentation on updating the SSL/TLS certificate](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL-certificate-rotation.html) for steps to rotate your certificate."
```

>   category: Admin

Reviewer 2: ```suggestion
  category: Search
```

  url: https://github.com/sourcegraph/sourcegraph/pull/18024
# Added

> - description: "New site config option `\"log\": { \"sentry\": { \"backendDSN\": \"\u003cREDACTED\u003e\" } }` to use a separate Sentry project for backend errors."

Reviewer: I recommend cutting this.

Reviewer 2: +1

> - description: "New site config option `\"log\": { \"sentry\": { \"backendDSN\": \"\u003cREDACTED\u003e\" } }` to use a separate Sentry project for backend errors."

Reviewer: I recommend cutting this from the post.

  url: https://github.com/sourcegraph/sourcegraph/pull/17363
  category:
- description: Structural search now supports searching indexed branches other than default.
  url: https://github.com/sourcegraph/sourcegraph/pull/17726
  category: Search

> - description: Structural search now supports searching unindexed revisions.

Reviewer: Would it make sense to combine the two structural search updates? 
"Structural search now supports searching unindexed revisions and indexed branches other than default."

Reviewer 2: +1 to combining these

  url: https://github.com/sourcegraph/sourcegraph/pull/17967
  category: Search

> - description: "New site config option `\"allowSignup\"` for SAML authentication to determine if automatically create new users is allowed."

Reviewer: I recommend removing this from the post.

  url: https://github.com/sourcegraph/sourcegraph/pull/17989
  category:

> - description: "Experimental: The webapp can now stream search results to the client, improving search performance. To enable it, add `{ \"experimentalFeatures\": { \"searchStreaming\": true } }` in user settings."

Reviewer: @christinaforney @content-hacker - how do you feel about including experimental features in the changelog?

Reviewer 4: It might be helpful to use different terminology. The changelog is what lives in our sourcegraph repo (what this was generated from) and should have everything added to the product. The blog version of the changelog is the distilled version of this with the 5-7 highlights for a release.

I think it can be fun to highlight experimental features - we can use it to get more users to turn on and test things for us, and shows incremental progress. We don't need to show every experimental thing though and should consider if this outcome is what we want for each thing.

Reviewer: @christinaforney - that was my mistake. I was referring to both the changelog and the release post as the "changelog." I was referring to including this in the release post.

Reviewer 4: I thought so :) my second paragraph hopefully answers that question for you?

Reviewer: Yes - thank you!

Reviewer: I like this.

@content-hacker - any recommendation for indicating that it's experimental?

Reviewer 2: I think we should keep it. Is experimental the right word? Is it too small of a change to use `beta` or is it too jarring to use a CTA (Try now: XYZ)

Just thoughts.. I have no problem with keeping experimental.

  url: https://github.com/sourcegraph/sourcegraph/pull/16097
  category: Search

> - description: "New product research sign-up page. This can be accessed by all users in their user settings."

Reviewer: ```suggestion
- description: "New product research sign-up page added to user settings. Users can now opt in to participate in user research and provide feedback ."
```

  url: https://github.com/sourcegraph/sourcegraph/pull/17945

>   category: Search

Reviewer: ```suggestion
  category: Admin
```

> - description: "New site config option `productResearchPage.enabled` to disable access to the product research sign-up page."

Reviewer: @content-hacker - we should either cut this one or mention it with the other product research page announcement ("Admins can choose to disable this page."

Reviewer 2: +1 to mentioning it with the product research page. I think it's easier for the reader to get all the information in one place.

  url: https://github.com/sourcegraph/sourcegraph/pull/17945
  category: Search

> - description: Pings now contain Sourcegraph extension activation statistics.

Reviewer: I recommend cutting this.

Reviewer 2: I'm seeing three lines for `Pings now contain...` in the preview. Can they be consolidated into one? I don't know enough about the feature/impact on user to say if we should cut them or not - your call!

  url: https://github.com/sourcegraph/sourcegraph/pull/16421

>   category:

Reviewer: I recommend cutting.

> - description: "Pings now contain aggregate Sourcegraph extension activation statistics: the number of users and number of activations per (public) extension per week, and the number of total extension users per week and average extensions activated per user."

Reviewer: I recommend cutting.

  url: https://github.com/sourcegraph/sourcegraph/pull/16421
  category:

> - description: "Pings now contain aggregate code insights usage data: total insight views, interactions, edits, creations, removals, and counts of unique users that view and create insights."

Reviewer: I recommend cutting.

  url: https://github.com/sourcegraph/sourcegraph/pull/17805
  category:
# Changed

> - description: "Alert solutions links included in [monitoring alerts](https://docs.sourcegraph.com/admin/observability/alerting) now link to the relevant documentation version."

Reviewer: I recommend cutting this.

  url: https://github.com/sourcegraph/sourcegraph/pull/17828
  category: Admin
- description: "Secrets (such as access tokens and passwords) will now appear as REDACTED when editing external service config, and in graphql API responses."
  url: https://github.com/sourcegraph/sourcegraph/issues/17261
  category: Admin
# Fixed

> - description: Fixes an issue that prevented the hard deletion of a user if they had saved searches.

Reviewer: I recommend cutting.

  url: https://github.com/sourcegraph/sourcegraph/pull/17461
  category: Search

> - description: "Fixes an issue that caused some missing results for `type:commit` when a pattern was used instead of the `message` field."

Reviewer: I recommend cutting.

  url: "https://github.com/sourcegraph/sourcegraph/pull/17490#issuecomment-764004758"
  category:

> - description: "Fixes an issue where cAdvisor-based alerts would not fire correctly for services with multiple replicas."

Reviewer: I recommend cutting.

  url: https://github.com/sourcegraph/sourcegraph/pull/17600
  category:

> - description: Significantly improved performance of structural search on monorepo deployments

Reviewer: ```suggestion
- description: Improved performance of structural search on monorepo deployments.
```

  url: https://github.com/sourcegraph/sourcegraph/pull/17846
  category: Search

> - description: Fixes an issue where upgrades on Kubernetes may fail due to null environment variable lists in deployment manifests

Reviewer: I recommend cutting this.

  url: "https://github.com/sourcegraph/deploy-sourcegraph/pull/1781"
  category:

> - description: Fixes an issue where counts on search filters were inaccurate.

Reviewer: I recommend cutting this
